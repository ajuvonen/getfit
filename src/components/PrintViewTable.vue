<script setup lang="ts">
defineProps<{
  tableTitle: string;
}>();
</script>
<template>
  <v-table class="print-view__table bg-transparent">
    <template #top>
      <h2 class="text-h5 pl-4">{{ tableTitle }}</h2>
    </template>
    <thead>
      <slot name="header"></slot>
    </thead>
    <tbody>
      <slot name="body"></slot>
    </tbody>
  </v-table>
</template>
<style lang="scss">
@media screen {
  .print-view__table + .print-view__table {
    margin-top: 2rem;
  }
}

@media print {
  .print-view__table {
    page-break-after: always;
    break-after: page;
  }
}
</style>
