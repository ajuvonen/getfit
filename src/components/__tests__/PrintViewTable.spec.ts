import {mount} from '@vue/test-utils';
import {describe, it, expect} from 'vitest';
import PrintViewTable from '@/components/PrintViewTable.vue';

describe('PrintViewTable', () => {
  it('mounts', () => {
    const wrapper = mount(PrintViewTable, {
      props: {
        tableTitle: 'My table',
      },
      slots: {
        header: '<tr><th>My header</th></tr>',
        body: '<tr><td>Body content</td></tr>',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('slot overrides table title', () => {
    const wrapper = mount(PrintViewTable, {
      props: {
        tableTitle: 'My table',
      },
      slots: {
        title: '<h2>My title</h2>',
        header: '<tr><th>My header</th></tr>',
        body: '<tr><td>Body content</td></tr>',
      },
    });
    expect(wrapper.find('h2').text()).toBe('My title');
  });

  it('uses striped prop', async () => {
    const wrapper = mount(PrintViewTable, {
      props: {
        striped: false,
      },
      slots: {
        header: '<tr><th>My header</th></tr>',
        body: '<tr><td>Body content</td></tr>',
      },
    });

    expect(wrapper.find('.print-view__table').classes()).to.not.contain(
      'print-view__table--striped',
    );
    await wrapper.setProps({striped: true});
    expect(wrapper.find('.print-view__table').classes()).to.contain('print-view__table--striped');
  });
});
