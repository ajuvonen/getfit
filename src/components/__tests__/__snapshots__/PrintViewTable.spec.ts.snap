// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PrintViewTable > mounts 1`] = `
"<div class=\\"v-table v-table--has-top v-theme--light v-table--density-default print-view__table bg-transparent\\">
  <h2 class=\\"text-h5 pl-4\\">My table</h2>
  <div class=\\"v-table__wrapper\\">
    <table>
      <thead>
        <tr>
          <th>My header</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>Body content</td>
        </tr>
      </tbody>
    </table>
  </div>
  <!---->
</div>"
`;
